blog-api-y-nguyen/Controllers/AuthorsController.cs

```diff
@@ -71,38 +75,38 @@ public IActionResult PutAuthor(int id, Author author)
                     throw;
                 }
             }
-            return NoContent();
+            return Ok();
         }
 
         // POST: api/Authors
         [HttpPost]
-        public ActionResult<Author> PostBlog(Author author)
+        public ActionResult<Author> PostAuthor(Author author)
         {
-            if (_authorRepository.CheckAuthorsIsNull())
+            if (_authorService.CheckAuthorsExist() == false)
             {
                 return Problem("Entity set 'BlogContext.Authors'  is null.");
             }
-            _authorRepository.PostAuthor(author);
-            _authorRepository.Save();
+            _authorService.PostAuthor(author);
+            _authorService.Save();
             return CreatedAtAction(nameof(GetAuthor), new { id = author.AuthorId }, author);
         }
 
         // DELETE: api/Authors/5
         [HttpDelete("{id}")]
         public IActionResult DeleteAuthor(int id)
         {
-            if (_authorRepository.CheckAuthorsIsNull())
+            if (_authorService.CheckAuthorsExist() == false)
             {
                 return NotFound();
             }
-            var authorDel = _authorRepository.GetAuthor(id);
+            var authorDel = _authorService.GetAuthor(id);
```

Comment:
bad naming `authorDel`, `author` is enough. Or `AuthorToBeDeleted`, do not scared to name a long variable name, as long as the user could easily figure out what it means